In [2]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
from ydata_profiling import ProfileReport

print("Environment setup successful!")

Environment setup successful!


C:\Users\local_ksalyamova\Temp\ipykernel_20116\892773448.py:5: DeprecationWarning: 
    `import ydata_profiling` is deprecated and will not receive more updates. 
    Please install fg-data-profiling via `pip install fg-data-profiling` and use `import data_profiling` instead.
    
  from ydata_profiling import ProfileReport


In [3]:
df = pd.read_csv("../data/neighbourhood-profiles-2016-140-model.csv")

In [4]:
print("Dataset Shape:")
print(df.shape)

print("\nColumn Names:")
print(df.columns.tolist())

print("\nFirst 5 rows:")
print(df.head())

Dataset Shape:
(2383, 146)

Column Names:
['_id', 'Category', 'Topic', 'Data Source', 'Characteristic', 'City of Toronto', 'Agincourt North', 'Agincourt South-Malvern West', 'Alderwood', 'Annex', 'Banbury-Don Mills', 'Bathurst Manor', 'Bay Street Corridor', 'Bayview Village', 'Bayview Woods-Steeles', 'Bedford Park-Nortown', 'Beechborough-Greenbrook', 'Bendale', 'Birchcliffe-Cliffside', 'Black Creek', 'Blake-Jones', 'Briar Hill-Belgravia', 'Bridle Path-Sunnybrook-York Mills', 'Broadview North', 'Brookhaven-Amesbury', 'Cabbagetown-South St. James Town', 'Caledonia-Fairbank', 'Casa Loma', 'Centennial Scarborough', 'Church-Yonge Corridor', 'Clairlea-Birchmount', 'Clanton Park', 'Cliffcrest', 'Corso Italia-Davenport', 'Danforth', 'Danforth East York', 'Don Valley Village', 'Dorset Park', 'Dovercourt-Wallace Emerson-Junction', 'Downsview-Roding-CFB', 'Dufferin Grove', 'East End-Danforth', 'Edenbridge-Humber Valley', 'Eglinton East', 'Elms-Old Rexdale', 'Englemount-Lawrence', 'Eringate-Centen

In [5]:
print("\nData Types Summary:")
print(df.dtypes.value_counts())

print("\nDetailed Data Types:")
print(df.dtypes)


Data Types Summary:
object    145
int64       1
Name: count, dtype: int64

Detailed Data Types:
_id                         int64
Category                   object
Topic                      object
Data Source                object
Characteristic             object
                            ...  
Wychwood                   object
Yonge-Eglinton             object
Yonge-St.Clair             object
York University Heights    object
Yorkdale-Glen Park         object
Length: 146, dtype: object


In [6]:
missing = df.isnull().sum()

print("Columns with Missing Values:")
print(missing[missing > 0].sort_values(ascending=False))

Columns with Missing Values:
Lawrence Park South               56
Pelmo Park-Humberlea              56
Oakridge                          56
Oakwood Village                   56
O'Connor-Parkview                 56
                                  ..
Eringate-Centennial-West Deane    56
Englemount-Lawrence               56
Elms-Old Rexdale                  56
Eglinton East                     56
City of Toronto                    7
Length: 141, dtype: int64


In [7]:
numeric_df = df.select_dtypes(include=['number'])

print("Number of Numeric Columns:")
print(numeric_df.shape)

print("\nNumeric Columns:")
print(numeric_df.columns.tolist())

Number of Numeric Columns:
(2383, 1)

Numeric Columns:
['_id']


In [8]:
ml_df = pd.read_csv("../data/neighbourhood_profiles_ml_ready.csv")

print(ml_df.shape)

print("\nData Types:")
print(ml_df.dtypes.value_counts())

print("\nFirst 5 rows:")
print(ml_df.head())

(143, 2384)

Data Types:
float64    2383
object        1
Name: count, dtype: int64

First 5 rows:
                     Unnamed: 0  \
0               City of Toronto   
1               Agincourt North   
2  Agincourt South-Malvern West   
3                     Alderwood   
4                         Annex   

   Neighbourhood Information | Neighbourhood Information | Neighbourhood Number  \
0                                                NaN                              
1                                              129.0                              
2                                              128.0                              
3                                               20.0                              
4                                               95.0                              

   Neighbourhood Information | Neighbourhood Information | TSNS2020 Designation  \
0                                                NaN                              
1                         

In [5]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2383 entries, 0 to 2382
Columns: 146 entries, _id to Yorkdale-Glen Park
dtypes: int64(1), object(145)
memory usage: 2.7+ MB


In [6]:
df.isnull().sum()

_id                         0
Category                    0
Topic                       0
Data Source                 0
Characteristic              0
                           ..
Wychwood                   56
Yonge-Eglinton             56
Yonge-St.Clair             56
York University Heights    56
Yorkdale-Glen Park         56
Length: 146, dtype: int64

In [7]:
df.describe(include='all')

,_id,Category,Topic,Data Source,Characteristic,City of Toronto,Agincourt North,Agincourt South-Malvern West,Alderwood,Annex,...,Willowdale West,Willowridge-Martingrove-Richview,Woburn,Woodbine Corridor,Woodbine-Lumsden,Wychwood,Yonge-Eglinton,Yonge-St.Clair,York University Heights,Yorkdale-Glen Park
count,2383.000000,2383,2383,2383,2383,2376,2327,2327,2327,2327,...,2327,2327,2327,2327,2327,2327,2327,2327,2327,2327
unique,NaN,15,49,5,1651,1527,662,650,522,693,...,611,627,811,522,453,542,507,532,702,552
top,NaN,Language,Ethnic origin population,Census Profile 98-316-X2016001,English and French,0,0,0,0,0,...,0,0,0,0,0,0,0,0,0,0
freq,NaN,808,279,2245,4,322,985,945,1030,818,...,920,855,769,987,1035,971,973,968,773,937
mean,1192.000000,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
std,688.057168,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
min,1.000000,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
25%,596.500000,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
50%,1192.000000,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
75%,1787.500000,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN


In [8]:
profile = ProfileReport(df, title="Toronto Neighbourhood Profiles EDA Report")

In [1]:
profile.to_file("../reports/Toronto_EDA_Report.html")

NameError: name 'profile' is not defined

In [2]:
import pandas as pd

df = pd.read_csv("../data/neighbourhood-profiles-2016-140-model.csv")

print(df.shape)

(2383, 146)


In [6]:
df.head()

,_id,Category,Topic,Data Source,Characteristic,City of Toronto,Agincourt North,Agincourt South-Malvern West,Alderwood,Annex,...,Willowdale West,Willowridge-Martingrove-Richview,Woburn,Woodbine Corridor,Woodbine-Lumsden,Wychwood,Yonge-Eglinton,Yonge-St.Clair,York University Heights,Yorkdale-Glen Park
0,1,Neighbourhood Information,Neighbourhood Information,City of Toronto,Neighbourhood Number,NaN,129,128,20,95,...,37,7,137,64,60,94,100,97,27,31
1,2,Neighbourhood Information,Neighbourhood Information,City of Toronto,TSNS2020 Designation,NaN,No Designation,No Designation,No Designation,No Designation,...,No Designation,No Designation,NIA,No Designation,No Designation,No Designation,No Designation,No Designation,NIA,Emerging Neighbourhood
2,3,Population,Population and dwellings,Census Profile 98-316-X2016001,"Population, 2016","2,731,571","29,113","23,757","12,054","30,526",...,"16,936","22,156","53,485","12,541","7,865","14,349","11,817","12,528","27,593","14,804"
3,4,Population,Population and dwellings,Census Profile 98-316-X2016001,"Population, 2011","2,615,060","30,279","21,988","11,904","29,177",...,"15,004","21,343","53,350","11,703","7,826","13,986","10,578","11,652","27,713","14,687"
4,5,Population,Population and dwellings,Census Profile 98-316-X2016001,Population Change 2011-2016,4.50%,-3.90%,8.00%,1.30%,4.60%,...,12.90%,3.80%,0.30%,7.20%,0.50%,2.60%,11.70%,7.50%,-0.40%,0.80%


In [7]:
df.columns.tolist()

['_id',
 'Category',
 'Topic',
 'Data Source',
 'Characteristic',
 'City of Toronto',
 'Agincourt North',
 'Agincourt South-Malvern West',
 'Alderwood',
 'Annex',
 'Banbury-Don Mills',
 'Bathurst Manor',
 'Bay Street Corridor',
 'Bayview Village',
 'Bayview Woods-Steeles',
 'Bedford Park-Nortown',
 'Beechborough-Greenbrook',
 'Bendale',
 'Birchcliffe-Cliffside',
 'Black Creek',
 'Blake-Jones',
 'Briar Hill-Belgravia',
 'Bridle Path-Sunnybrook-York Mills',
 'Broadview North',
 'Brookhaven-Amesbury',
 'Cabbagetown-South St. James Town',
 'Caledonia-Fairbank',
 'Casa Loma',
 'Centennial Scarborough',
 'Church-Yonge Corridor',
 'Clairlea-Birchmount',
 'Clanton Park',
 'Cliffcrest',
 'Corso Italia-Davenport',
 'Danforth',
 'Danforth East York',
 'Don Valley Village',
 'Dorset Park',
 'Dovercourt-Wallace Emerson-Junction',
 'Downsview-Roding-CFB',
 'Dufferin Grove',
 'East End-Danforth',
 'Edenbridge-Humber Valley',
 'Eglinton East',
 'Elms-Old Rexdale',
 'Englemount-Lawrence',
 'Eringate-Ce

In [9]:
df.columns.tolist()[:10]

['_id',
 'Category',
 'Topic',
 'Data Source',
 'Characteristic',
 'City of Toronto',
 'Agincourt North',
 'Agincourt South-Malvern West',
 'Alderwood',
 'Annex']

In [10]:
df["Indicator"] = (
    df["Category"].astype(str) + " | " +
    df["Topic"].astype(str) + " | " +
    df["Characteristic"].astype(str))

df["Indicator"].head()

0    Neighbourhood Information | Neighbourhood Info...
1    Neighbourhood Information | Neighbourhood Info...
2    Population | Population and dwellings | Popula...
3    Population | Population and dwellings | Popula...
4    Population | Population and dwellings | Popula...
Name: Indicator, dtype: object

In [11]:
df["Indicator"].nunique()

2245

In [12]:
neighbourhood_df = df.drop(
    columns=["_id", "Category", "Topic", "Data Source", "Characteristic"])

neighbourhood_df = neighbourhood_df.set_index("Indicator")

ml_df = neighbourhood_df.T

print(ml_df.shape)

(141, 2383)


In [13]:
duplicates = df["Indicator"].duplicated().sum()

print("Duplicate indicators:", duplicates)

Duplicate indicators: 138


In [14]:
df[df["Indicator"].duplicated(keep=False)][
    ["Category", "Topic", "Characteristic", "Data Source"]
].head(20)

,Category,Topic,Characteristic,Data Source
268,Income,Income of households in 2015,"$150,000 to $199,999",Census Profile 98-316-X2016001
426,Income,Income of individuals in 2015,"$60,000 to $69,999",Census Profile 98-316-X2016001
615,Income,Income of individuals in 2015,"$70,000 to $79,999",Census Profile 98-316-X2016001
839,Income,Income of individuals in 2015,"$100,000 and over",Census Profile 98-316-X2016001
950,Income,Income of individuals in 2015,"$30,000 to $39,999",Census Profile 98-316-X2016001
957,Income,Income of individuals in 2015,"$40,000 to $49,999",Census Profile 98-316-X2016001
964,Income,Income of individuals in 2015,"$50,000 to $59,999",Census Profile 98-316-X2016001
981,Income,Income of individuals in 2015,"Under $10,000 (including loss)",Census Profile 98-316-X2016001
982,Income,Income of individuals in 2015,"$10,000 to $19,999",Census Profile 98-316-X2016001
983,Income,Income of individuals in 2015,"$20,000 to $29,999",Census Profile 98-316-X2016001


In [15]:
dup_rows = df[df["Indicator"].duplicated(keep=False)]

dup_rows[["Indicator"]].head(20)

,Indicator
268,Income | Income of households in 2015 | $1...
426,Income | Income of individuals in 2015 | $...
615,Income | Income of individuals in 2015 | $...
839,Income | Income of individuals in 2015 | ...
950,Income | Income of individuals in 2015 | $...
957,Income | Income of individuals in 2015 | $...
964,Income | Income of individuals in 2015 | $...
981,Income | Income of individuals in 2015 | U...
982,Income | Income of individuals in 2015 | $...
983,Income | Income of individuals in 2015 | $...


In [16]:
dup_rows["Indicator"].value_counts().head(20)

Indicator
Income | Low income in 2015 |     0 to 5 years                                                                                                                       3
Income | Income of individuals in 2015 |     $70,000 to $79,999                                                                                                      3
Income | Income of individuals in 2015 |     $30,000 to $39,999                                                                                                      3
Income | Income of individuals in 2015 |     $40,000 to $49,999                                                                                                      3
Income | Income of individuals in 2015 |     $50,000 to $59,999                                                                                                      3
Income | Low income in 2015 |   65 years and over                                                                                                          

In [19]:
dup_v2 = df[df["Indicator_v2"].duplicated(keep=False)]

dup_v2[["Indicator_v2", "Data Source"]].head(20)

,Indicator_v2,Data Source
268,"Income | Income of households in 2015 | $150,000 to $199,999",Census Profile 98-316-X2016001
426,"Income | Income of individuals in 2015 | $60,000 to $69,999",Census Profile 98-316-X2016001
615,"Income | Income of individuals in 2015 | $70,000 to $79,999",Census Profile 98-316-X2016001
839,"Income | Income of individuals in 2015 | $100,000 and over",Census Profile 98-316-X2016001
950,"Income | Income of individuals in 2015 | $30,000 to $39,999",Census Profile 98-316-X2016001
957,"Income | Income of individuals in 2015 | $40,000 to $49,999",Census Profile 98-316-X2016001
964,"Income | Income of individuals in 2015 | $50,000 to $59,999",Census Profile 98-316-X2016001
981,"Income | Income of individuals in 2015 | Under $10,000 (including loss)",Census Profile 98-316-X2016001
982,"Income | Income of individuals in 2015 | $10,000 to $19,999",Census Profile 98-316-X2016001
983,"Income | Income of individuals in 2015 | $20,000 to $29,999",Census Profile 98-316-X2016001


In [20]:
dup_v2[["Category", "Topic", "Characteristic"]].loc[[981, 988]]

,Category,Topic,Characteristic
981,Income,Income of individuals in 2015,"Under $10,000 (including loss)"
988,Income,Income of individuals in 2015,"Under $10,000 (including loss)"


In [21]:
dup_v2[["Category", "Topic", "Characteristic"]].loc[[982, 989]]

,Category,Topic,Characteristic
982,Income,Income of individuals in 2015,"$10,000 to $19,999"
989,Income,Income of individuals in 2015,"$10,000 to $19,999"


In [22]:
df.loc[[981, 988]]

,_id,Category,Topic,Data Source,Characteristic,City of Toronto,Agincourt North,Agincourt South-Malvern West,Alderwood,Annex,...,Woburn,Woodbine Corridor,Woodbine-Lumsden,Wychwood,Yonge-Eglinton,Yonge-St.Clair,York University Heights,Yorkdale-Glen Park,Indicator,Indicator_v2
981,982,Income,Income of individuals in 2015,Census Profile 98-316-X2016001,"Under $10,000 (including loss)","389,430","5,170","4,535","1,365","3,870",...,"8,875","1,300",870,"1,630","1,290","1,195","5,105","1,835","Income | Income of individuals in 2015 | Under $10,000 (including loss)","Income | Income of individuals in 2015 | Under $10,000 (including loss)"
988,989,Income,Income of individuals in 2015,Census Profile 98-316-X2016001,"Under $10,000 (including loss)","395,780","5,225","4,590","1,395","3,995",...,"8,955","1,320",885,"1,670","1,325","1,275","5,135","1,875","Income | Income of individuals in 2015 | Under $10,000 (including loss)","Income | Income of individuals in 2015 | Under $10,000 (including loss)"


In [23]:
df.loc[[981, 988], ["_id", "Category", "Topic", "Characteristic", "City of Toronto"]]

,_id,Category,Topic,Characteristic,City of Toronto
981,982,Income,Income of individuals in 2015,"Under $10,000 (including loss)","389,430"
988,989,Income,Income of individuals in 2015,"Under $10,000 (including loss)","395,780"


In [24]:
(df.loc[981] == df.loc[988]).all()

False

In [25]:
different_cols = df.columns[df.loc[981] != df.loc[988]]

print(different_cols.tolist())

['_id', 'City of Toronto', 'Agincourt North', 'Agincourt South-Malvern West', 'Alderwood', 'Annex', 'Banbury-Don Mills', 'Bathurst Manor', 'Bay Street Corridor', 'Bayview Village', 'Bayview Woods-Steeles', 'Bedford Park-Nortown', 'Beechborough-Greenbrook', 'Bendale', 'Birchcliffe-Cliffside', 'Black Creek', 'Blake-Jones', 'Briar Hill-Belgravia', 'Bridle Path-Sunnybrook-York Mills', 'Broadview North', 'Brookhaven-Amesbury', 'Cabbagetown-South St. James Town', 'Caledonia-Fairbank', 'Casa Loma', 'Centennial Scarborough', 'Church-Yonge Corridor', 'Clairlea-Birchmount', 'Clanton Park', 'Cliffcrest', 'Corso Italia-Davenport', 'Danforth', 'Danforth East York', 'Don Valley Village', 'Dorset Park', 'Dovercourt-Wallace Emerson-Junction', 'Downsview-Roding-CFB', 'Dufferin Grove', 'East End-Danforth', 'Edenbridge-Humber Valley', 'Eglinton East', 'Elms-Old Rexdale', 'Englemount-Lawrence', 'Eringate-Centennial-West Deane', 'Etobicoke West Mall', 'Flemingdon Park', 'Forest Hill North', 'Forest Hill So

In [26]:
neighbourhood_df = df.drop(columns=["_id", "Category", "Topic", "Data Source", "Characteristic"])

ml_df = neighbourhood_df.T

print(ml_df.shape)

(143, 2383)


In [27]:
ml_df.dtypes.value_counts()

object    2383
Name: count, dtype: int64

In [28]:
ml_df.head()

,0,1,2,3,4,5,6,7,8,9,...,2373,2374,2375,2376,2377,2378,2379,2380,2381,2382
City of Toronto,NaN,NaN,"2,731,571","2,615,060",4.50%,"1,179,057","1,112,929","4,334",630.2,"398,135",...,"49,990","15,780","59,945","2,556,120","1,516,110","1,040,015","639,060","400,950","42,985","216,835"
Agincourt North,129,No Designation,"29,113","30,279",-3.90%,"9,371","9,120","3,929",7.41,"3,840",...,275,75,605,"27,490","18,865","8,610","5,445","3,170",135,"2,280"
Agincourt South-Malvern West,128,No Designation,"23,757","21,988",8.00%,"8,535","8,136","3,034",7.83,"3,075",...,320,90,490,"22,325","13,565","8,775","5,610","3,145",220,"2,170"
Alderwood,20,No Designation,"12,054","11,904",1.30%,"4,732","4,616","2,435",4.95,"1,760",...,220,40,70,"11,370","8,235","3,130","2,200",925,70,245
Annex,95,No Designation,"30,526","29,177",4.60%,"18,109","15,934","10,863",2.81,"2,360",...,900,385,835,"27,715","12,980","14,735","8,340","6,390","1,310","2,460"


In [29]:
ml_df.columns = df["Indicator"]

ml_df.head()

Indicator,Neighbourhood Information | Neighbourhood Information | Neighbourhood Number,Neighbourhood Information | Neighbourhood Information | TSNS2020 Designation,"Population | Population and dwellings | Population, 2016","Population | Population and dwellings | Population, 2011",Population | Population and dwellings | Population Change 2011-2016,Population | Population and dwellings | Total private dwellings,Population | Population and dwellings | Private dwellings occupied by usual residents,Population | Population and dwellings | Population density per square kilometre,Population | Population and dwellings | Land area in square kilometres,Population | Age characteristics | Children (0-14 years),...,Mobility | Mobility status - Place of residence 1 year ago | Intraprovincial migrants,Mobility | Mobility status - Place of residence 1 year ago | Interprovincial migrants,Mobility | Mobility status - Place of residence 1 year ago | External migrants,Mobility | Mobility status - Place of residence 5 years ago | Total - Mobility status 5 years ago - 25% sample data,Mobility | Mobility status - Place of residence 5 years ago | Non-movers,Mobility | Mobility status - Place of residence 5 years ago | Movers,Mobility | Mobility status - Place of residence 5 years ago | Non-migrants,Mobility | Mobility status - Place of residence 5 years ago | Migrants,Mobility | Mobility status - Place of residence 5 years ago | Interprovincial migrants,Mobility | Mobility status - Place of residence 5 years ago | External migrants
City of Toronto,NaN,NaN,"2,731,571","2,615,060",4.50%,"1,179,057","1,112,929","4,334",630.2,"398,135",...,"49,990","15,780","59,945","2,556,120","1,516,110","1,040,015","639,060","400,950","42,985","216,835"
Agincourt North,129,No Designation,"29,113","30,279",-3.90%,"9,371","9,120","3,929",7.41,"3,840",...,275,75,605,"27,490","18,865","8,610","5,445","3,170",135,"2,280"
Agincourt South-Malvern West,128,No Designation,"23,757","21,988",8.00%,"8,535","8,136","3,034",7.83,"3,075",...,320,90,490,"22,325","13,565","8,775","5,610","3,145",220,"2,170"
Alderwood,20,No Designation,"12,054","11,904",1.30%,"4,732","4,616","2,435",4.95,"1,760",...,220,40,70,"11,370","8,235","3,130","2,200",925,70,245
Annex,95,No Designation,"30,526","29,177",4.60%,"18,109","15,934","10,863",2.81,"2,360",...,900,385,835,"27,715","12,980","14,735","8,340","6,390","1,310","2,460"


In [31]:
ml_df_clean = ml_df.replace(",", "", regex=True)
ml_df_clean = ml_df_clean.replace("%", "", regex=True)

ml_df_numeric = ml_df_clean.apply(pd.to_numeric, errors="coerce")

print(ml_df_numeric.shape)
print(ml_df_numeric.dtypes.value_counts())

(143, 2383)
float64    2383
Name: count, dtype: int64


In [32]:
ml_df_numeric.info()

<class 'pandas.core.frame.DataFrame'>
Index: 143 entries, City of Toronto to Indicator_v2
Columns: 2383 entries, Neighbourhood Information | Neighbourhood Information | Neighbourhood Number to Mobility | Mobility status - Place of residence 5 years ago |       External migrants
dtypes: float64(2383)
memory usage: 2.6+ MB


In [33]:
ml_df_numeric.head()

Indicator,Neighbourhood Information | Neighbourhood Information | Neighbourhood Number,Neighbourhood Information | Neighbourhood Information | TSNS2020 Designation,"Population | Population and dwellings | Population, 2016","Population | Population and dwellings | Population, 2011",Population | Population and dwellings | Population Change 2011-2016,Population | Population and dwellings | Total private dwellings,Population | Population and dwellings | Private dwellings occupied by usual residents,Population | Population and dwellings | Population density per square kilometre,Population | Population and dwellings | Land area in square kilometres,Population | Age characteristics | Children (0-14 years),...,Mobility | Mobility status - Place of residence 1 year ago | Intraprovincial migrants,Mobility | Mobility status - Place of residence 1 year ago | Interprovincial migrants,Mobility | Mobility status - Place of residence 1 year ago | External migrants,Mobility | Mobility status - Place of residence 5 years ago | Total - Mobility status 5 years ago - 25% sample data,Mobility | Mobility status - Place of residence 5 years ago | Non-movers,Mobility | Mobility status - Place of residence 5 years ago | Movers,Mobility | Mobility status - Place of residence 5 years ago | Non-migrants,Mobility | Mobility status - Place of residence 5 years ago | Migrants,Mobility | Mobility status - Place of residence 5 years ago | Interprovincial migrants,Mobility | Mobility status - Place of residence 5 years ago | External migrants
City of Toronto,NaN,NaN,2731571.0,2615060.0,4.5,1179057.0,1112929.0,4334.0,630.20,398135.0,...,49990.0,15780.0,59945.0,2556120.0,1516110.0,1040015.0,639060.0,400950.0,42985.0,216835.0
Agincourt North,129.0,NaN,29113.0,30279.0,-3.9,9371.0,9120.0,3929.0,7.41,3840.0,...,275.0,75.0,605.0,27490.0,18865.0,8610.0,5445.0,3170.0,135.0,2280.0
Agincourt South-Malvern West,128.0,NaN,23757.0,21988.0,8.0,8535.0,8136.0,3034.0,7.83,3075.0,...,320.0,90.0,490.0,22325.0,13565.0,8775.0,5610.0,3145.0,220.0,2170.0
Alderwood,20.0,NaN,12054.0,11904.0,1.3,4732.0,4616.0,2435.0,4.95,1760.0,...,220.0,40.0,70.0,11370.0,8235.0,3130.0,2200.0,925.0,70.0,245.0
Annex,95.0,NaN,30526.0,29177.0,4.6,18109.0,15934.0,10863.0,2.81,2360.0,...,900.0,385.0,835.0,27715.0,12980.0,14735.0,8340.0,6390.0,1310.0,2460.0


In [34]:
ml_df_numeric.to_csv("../data/neighbourhood_profiles_ml_ready.csv")

In [35]:
import pandas as pd

ml_df = pd.read_csv("../data/neighbourhood_profiles_ml_ready.csv")

print(ml_df.shape)
ml_df.head()

(143, 2384)


,Unnamed: 0,Neighbourhood Information | Neighbourhood Information | Neighbourhood Number,Neighbourhood Information | Neighbourhood Information | TSNS2020 Designation,"Population | Population and dwellings | Population, 2016","Population | Population and dwellings | Population, 2011",Population | Population and dwellings | Population Change 2011-2016,Population | Population and dwellings | Total private dwellings,Population | Population and dwellings | Private dwellings occupied by usual residents,Population | Population and dwellings | Population density per square kilometre,Population | Population and dwellings | Land area in square kilometres,...,Mobility | Mobility status - Place of residence 1 year ago | Intraprovincial migrants,Mobility | Mobility status - Place of residence 1 year ago | Interprovincial migrants,Mobility | Mobility status - Place of residence 1 year ago | External migrants,Mobility | Mobility status - Place of residence 5 years ago | Total - Mobility status 5 years ago - 25% sample data,Mobility | Mobility status - Place of residence 5 years ago | Non-movers,Mobility | Mobility status - Place of residence 5 years ago | Movers,Mobility | Mobility status - Place of residence 5 years ago | Non-migrants,Mobility | Mobility status - Place of residence 5 years ago | Migrants,Mobility | Mobility status - Place of residence 5 years ago | Interprovincial migrants,Mobility | Mobility status - Place of residence 5 years ago | External migrants
0,City of Toronto,NaN,NaN,2731571.0,2615060.0,4.5,1179057.0,1112929.0,4334.0,630.20,...,49990.0,15780.0,59945.0,2556120.0,1516110.0,1040015.0,639060.0,400950.0,42985.0,216835.0
1,Agincourt North,129.0,NaN,29113.0,30279.0,-3.9,9371.0,9120.0,3929.0,7.41,...,275.0,75.0,605.0,27490.0,18865.0,8610.0,5445.0,3170.0,135.0,2280.0
2,Agincourt South-Malvern West,128.0,NaN,23757.0,21988.0,8.0,8535.0,8136.0,3034.0,7.83,...,320.0,90.0,490.0,22325.0,13565.0,8775.0,5610.0,3145.0,220.0,2170.0
3,Alderwood,20.0,NaN,12054.0,11904.0,1.3,4732.0,4616.0,2435.0,4.95,...,220.0,40.0,70.0,11370.0,8235.0,3130.0,2200.0,925.0,70.0,245.0
4,Annex,95.0,NaN,30526.0,29177.0,4.6,18109.0,15934.0,10863.0,2.81,...,900.0,385.0,835.0,27715.0,12980.0,14735.0,8340.0,6390.0,1310.0,2460.0


In [36]:
ml_df.dtypes.value_counts()

float64    2383
object        1
Name: count, dtype: int64

In [10]:
from ydata_profiling import ProfileReport

profile = ProfileReport(ml_df,
    title="ML Ready Toronto Neighbourhood Dataset")

profile.to_file("../reports/Toronto_EDA_Revised.html")

Summarize dataset:   0%|          | 0/5 [00:00<?, ?it/s]

100%|██████████| 2384/2384 [02:02<00:00, 19.51it/s]


In [39]:
profile = ProfileReport(df,
    title="Toronto Neighbourhood Profiles EDA",
    minimal=True)

profile.to_file("../reports/Toronto_EDA_Revised.html")

Summarize dataset:   0%|          | 0/5 [00:00<?, ?it/s]

100%|██████████| 148/148 [00:19<00:00,  7.46it/s]


Generate report structure:   0%|          | 0/1 [00:00<?, ?it/s]

Render HTML:   0%|          | 0/1 [00:00<?, ?it/s]

Export report to file:   0%|          | 0/1 [00:00<?, ?it/s]

In [40]:
import pandas as pd

ml_check = pd.read_csv("../data/neighbourhood_profiles_ml_ready.csv")

print(ml_check.shape)

(143, 2384)


## Machine Learning Dataset

The evolution of the dataset from the original EDA to revised EDA: The original Toronto Neighbourhood Profiles dataset contained 2,383 indicators in rows and  143 neighbourhoods in columns. After data transposing, we achieved to 143 observations in rows and 2383 neighboorhoods in columns (to float from object)

Data preparation included:
- Removed the metadata fields
- Identified duplicate indicators
- Created unique indicator names (Indicator_v2) 
- Generated revised version of the EDA report (Toronto_EDA_Revised.html)
- Transposed the dataset for machine learning
- Converted values to numeric format
- Created and exported the ML-ready dataset (neighborhood_profiles_ml_ready.csv)
- Documented the transformation process 

Final dataset dimensions:
- Observations (Neighbourhoods): 143
- Features (Indicators): 2,383

In [41]:
ml_df_numeric.describe().T.head(10)

,count,mean,std,min,25%,50%,75%,max
Indicator,,,,,,,,
Neighbourhood Information | Neighbourhood Information | Neighbourhood Number,140.0,70.500000,40.558600,1.00,35.75,70.50,105.25,140.0
Neighbourhood Information | Neighbourhood Information | TSNS2020 Designation,0.0,NaN,NaN,NaN,NaN,NaN,NaN,NaN
"Population | Population and dwellings | Population, 2016",141.0,38745.687943,228615.346854,6577.00,12054.00,16775.00,23925.00,2731571.0
"Population | Population and dwellings | Population, 2011",141.0,37093.049645,218842.688157,6488.00,11904.00,16423.00,22829.00,2615060.0
Population | Population and dwellings | Population Change 2011-2016,141.0,4.043972,8.871345,-7.60,-0.30,1.50,4.40,52.0
Population | Population and dwellings | Total private dwellings,141.0,16724.212766,98729.853714,2796.00,5386.00,6885.00,10111.00,1179057.0
Population | Population and dwellings | Private dwellings occupied by usual residents,141.0,15786.226950,93178.644646,2650.00,5168.00,6576.00,9558.00,1112929.0
Population | Population and dwellings | Population density per square kilometre,141.0,6247.468085,4825.770903,1040.00,3596.00,5070.00,7601.00,44321.0
Population | Population and dwellings | Land area in square kilometres,141.0,8.939291,52.887526,0.42,1.86,3.31,5.39,630.2


In [42]:
ml_check.columns[0]

'Unnamed: 0'

In [43]:
ml_check = pd.read_csv(
    "../data/neighbourhood_profiles_ml_ready.csv",
    index_col=0)

ml_check.shape

(143, 2383)

In [3]:
from ydata_profiling import ProfileReport